lib/hsa/mcwamp_hsa.cpp

```diff
@@ -3171,6 +3191,21 @@ HSADispatch::setLaunchConfiguration(int dims, size_t *globalDims, size_t *localD
       workgroup_total_size = workgroup_size[0] * workgroup_size[1] * workgroup_size[2];
     }
 
+    static const size_t num_work_items_per_simd = 64;
+    static const size_t num_simds_per_cu = 4;
+    static const size_t num_work_items_per_cu = num_work_items_per_simd * num_simds_per_cu;
+    size_t num = (workgroup_total_size + num_work_items_per_cu - 1) / num_work_items_per_cu;
+    size_t num_sgprs = num * kernel->wavefront_sgpr_count;
+    size_t num_vgprs = num * kernel->workitem_vgpr_count;
+    // Every wavefront is allocated up to a maximum of 104 SGPRs on the GPUs prior to VI and 102 SGPRs on VI.
+    if (num_sgprs > 102) {
+        throw Kalmar::runtime_exception("The number of SGPRs needed by this launch exceeds HW limit due to big work group size!", 0);
+    }
+    // Every work-item has access to some number of VGPRs, up to a maximum of 256.
+    if (num_vgprs > 256) {
```

Comment:
@weixingzhang could you revise this value as a macro, with comment to describe which ISA version does this value apply to (ex: GFX7 above , or only for GFX8).